### PR TITLE
MD Trajectory Analysis tools 

### PR DESCRIPTION
**Pull request by chrisbarnettster**
Hi everyone 
some updates and new tools 
1.  Updated Ramachandran tool, selection done using a YAML syntax (mdanalysis/ramachandran_plots.*)
2.  New Ramachandran tool for proteins (mdanalysis/ramachandran_auto_protein.*)
3. End to End distance timeseries and histogram for proteins (mdanalysis/end-to-end.*)
4. Trajectory tool which allows for selections and multijoin. (mdtraj/traj_select_and_merge.xml)

### CODE REVIEW

File: tools/mdanalysis/end-to-end.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+<tool id="mdanalysis_endtoend" name="End-to-End Analysis" version="@VERSION@+galaxy@GALAXY_VERSION@">
+	<description>- End-to-End distance timeseries and histogram for the given selections</description>
+	<macros>
+		<import>macros.xml</import>
```
Comment (bgruening) [inline tools/mdanalysis/end-to-end.xml:4]: indentation should be 4spaces and not tab

File: tools/mdanalysis/end-to-end.xml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		<token name="@GALAXY_VERSION@">0</token>
+	</macros>
+	<expand macro="requirements"/>
+	<command detect_errors="exit_code">
+		<![CDATA[
```
Comment (bgruening) [inline tools/mdanalysis/end-to-end.xml:9]: indentation ...

File: tools/mdanalysis/ramachandran_auto_protein.xml [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            --o_plot1 '$o_plot1'
+            --o_plot2 '$o_plot2'
+            --o_data1 '$o_data1'
+            --o_html1 '$o_html1'
+        2>&1
+]]>
+  </command>
+  <inputs>
+    <expand macro="analysis_inputs"/>
+    <param name="segid1" type="text" value="PROA" label="Segment ID of atom 1"/>
```
Comment (bgruening) [inline tools/mdanalysis/ramachandran_auto_protein.xml:40]: this needs a validator, I think we have in other tools already some which you can reuse.

File: tools/mdanalysis/ramachandran_auto_protein.xml [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+      <option value="id">ResidueID</option>
+    </param>
+    <conditional name="subselectionr">
+      <param name="residuenames" type="select" label="Refine with a resname selection" help="Choose residues to refine the selection">
+        <option selected="True" value="No">No</option>
+        <option value="Yes">Yes</option>
+      </param>
+      <when value="No"></when>
+      <when value="Yes">
+        <param name="resname" type="text" value="ALA" label="Resname"/>
```
Comment (bgruening) [inline tools/mdanalysis/ramachandran_auto_protein.xml:53]: can this be optional=True and then the conditional can be removed?

**Comment by chrisbarnettster** [comment]
OK I think I got this sorted. Removed the conditional. Cleaned up those awful tabs. Included validators. Any further comments @tsenapathi @bgruening ?

File: tools/mdanalysis/end-to-end.xml [hunk 5]
```diff
@@ -0,0 +1,91 @@
+<tool id="mdanalysis_endtoend" name="End-to-End Analysis" version="@VERSION@+galaxy@GALAXY_VERSION@">
+    <description>- End-to-End distance timeseries and histogram for the given selections</description>
+    <macros>
+        <import>macros.xml</import>
+        <token name="@GALAXY_VERSION@">0</token>
```
Comment (simonbray) [inline tools/mdanalysis/end-to-end.xml:5]: Nice :+1:

File: tools/mdanalysis/end-to-end.xml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        --isegid1 '$segid1'
+        --ilabel '$label1'
+        --ititle1 '$title1'
+        --output1 '$output1'
+        --o_plot '$o_plot'
+    2>&1
+]]></command>
+    <inputs>
+        <expand macro="analysis_inputs"/>
+        <param name="segid1" type="text" value="PROA" label="Segment ID of atom 1">
```
Comment (simonbray) [inline tools/mdanalysis/end-to-end.xml:24]: can you add some help text to these params?

File: tools/mdanalysis/ramachandran_auto_protein.xml [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+            #end if
+            --o_plot1 '$o_plot1'
+            --o_plot2 '$o_plot2'
+            --o_data1 '$o_data1'
+            --o_html1 '$o_html1'
+        2>&1
+]]></command>
+    <inputs>
+        <expand macro="analysis_inputs" />
+        <param name="segid1" type="text" value="PROA" label="Segment ID of atom 1">
```
Comment (simonbray) [inline tools/mdanalysis/ramachandran_auto_protein.xml:38]: why atom 1? should it be 'of the protein'?

**Comment by chrisbarnettster** [comment]
@simonbray Thanks updates made.

**Comment by chrisbarnettster** [comment]
@simonbray @bgruening could you squash and merge?

**Comment by bgruening** [comment]
Of course!

**Pull request merged**